src/wazuh_db/main.c

```diff
@@ -246,13 +244,12 @@ int main(int argc, char ** argv)
     // Router module logging initialization
     router_initialize(taggedLogFunction);
 
-    if (status = pthread_create(&thread_router, NULL, run_router_handle, NULL), status != 0) {
-        merror("Couldn't create 'run_router_handle' thread: %s", strerror(status));
-        goto failure;
+    // Router provider initialization
+    if (router_syscollector_handle = router_provider_create(WDB_SYSCOLLECTOR_DELTAS_TOPIC), !router_syscollector_handle) {
```

Comment:
Now that I see this again, it might be a good idea to move the initialization a few lines above, before the worker threads of `wazuh-db` are launched.

Reply:
good catch